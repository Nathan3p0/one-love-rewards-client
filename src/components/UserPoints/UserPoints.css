.user__content-info--graph {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background: #429398;
    padding: 3rem 0;
    border: 2px solid rgba(255, 255, 255, 0.322);
    box-shadow: 1px 3px 2px rgb(39, 39, 39);
  }


  .user__content-info--graph h3 {
    font-size: 2.5rem;
    color: #FDFFFC;
    text-shadow: 1px 1px 3px rgb(93, 93, 93);
  }

  .points-total p {
    font-size: 3rem;
    margin: 0;
    color: #FDFFFC;
    font-weight: 700;
    text-shadow: 1px 1px 3px black;
  }

  @media screen and (max-width: 600px) {
    .user__content-info--graph h3 {
      font-size: 1.8rem;
      padding: 1rem;
    }

    .user__content-info--graph {
      padding-top: 1rem;
    }
  }