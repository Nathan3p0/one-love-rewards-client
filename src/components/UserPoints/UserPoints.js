import React from 'react'
import './circle.css'
import './UserPoints.css'

const UserPoints = (props) => {
    const graphStyle = `c100 p${Math.floor(props.percentage * 100)} big dark`

    return (
        <section className="user__content-info--graph">
            <h3>Current Point Status:</h3>
            <div className={graphStyle}>
                <span>{Math.floor(props.percentage * 100)}%</span>
                <div className="slice">
                    <div className="bar"></div>
                    <div className="fill"></div>
                </div>
            </div>
            <div className="points-total">
                <p>{props.points}pts</p>
            </div>
            <p>**Percentage to your next reward**</p>
        </section>
    );
}

UserPoints.defaultProps = {
    rewards: []
}

export default UserPoints;