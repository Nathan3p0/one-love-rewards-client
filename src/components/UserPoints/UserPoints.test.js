import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import UserPoints from './UserPoints'

describe('User Points Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<UserPoints />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<UserPoints />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<UserPoints points={12} percentage={50} rewards={[]} />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})