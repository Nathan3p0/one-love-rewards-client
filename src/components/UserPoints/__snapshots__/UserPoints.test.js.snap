// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="user__content-info--graph"
>
  <h3>
    Current Point Status
  </h3>
  <div
    className="c100 pNaN big dark"
  >
    <span>
      NaN
      %
    </span>
    <div
      className="slice"
    >
      <div
        className="bar"
      />
      <div
        className="fill"
      />
    </div>
  </div>
  <div
    className="points-total"
  >
    <p>
      pts
    </p>
  </div>
</section>
`;

exports[`Card Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<section
  className="user__content-info--graph"
>
  <h3>
    Current Point Status
  </h3>
  <div
    className="c100 p5000 big dark"
  >
    <span>
      5000
      %
    </span>
    <div
      className="slice"
    >
      <div
        className="bar"
      />
      <div
        className="fill"
      />
    </div>
  </div>
  <div
    className="points-total"
  >
    <p>
      12
      pts
    </p>
  </div>
</section>
`;
